pkg/config/config.go

```diff
@@ -204,6 +204,7 @@ func InitConfig(config Config) {
 	// Debugging + C-land crash feature flags
 	config.BindEnvAndSetDefault("c_stacktrace_collection", false)
 	config.BindEnvAndSetDefault("c_core_dump", false)
+	config.BindEnvAndSetDefault("go_core_dump", false)
```

Comment:
`c_core_dump` was originally implemented to produce a core dump in case of `SIGSEGV` (with the C code of rtloader in mind, but in reality, the signal handler wouldn’t care what is the source of `SIGSEGV`)

The new `go_core_dump` seems more generic as its goal is to produce a core dump in case of panic whatever the reason for the panic.

If I’m not wrong, by default, a `SIGSEGV` is caught by the GO runtime and turned into a panic.

So, is it correct to say that `go_core_dump` obsoletes `c_core_dump` and that the two features could be merged and renamed `core_dump` ?

Reply:
> If I’m not wrong, by default, a SIGSEGV is caught by the GO runtime and turned into a panic.

I think by default it is not caught by the Go runtime (couldn't find a docs link but see e.g. [here](https://keyanp.com/cgo-segfault.html)), so we still need the `c_core_dump` option to enable it; what this option does is install a signal handler that prints the C stack on SIGSEV:
https://github.com/DataDog/datadog-agent/blob/8eee7f164962897ba4f8f979775e0d8ad5adb6f6/rtloader/rtloader/api.cpp#L373-L384

I would prefer to keep the two separate but I am fine either way.

Reply:
I just made a try and `SIGSEGV` are caught by the GO runtime.
But the message prints only the GO part of the stack.
And printing the C++ part of the stack in case of segmentation fault in rtloader is definitively valuable.